nipype/interfaces/dtitk/registration.py

```diff
@@ -1,394 +1,419 @@
+    >>> node.inputs.in_aff= 'im2.nii'
+    >>> node.cmdline
+    'dfRightComposeAffine -aff im2.nii -df im1.nii -out im1_aff.df'
+    >>> node.run() # doctest: +SKIP
+    """
     input_spec = ComposeXfmInputSpec
     output_spec = ComposeXfmOutputSpec
     _cmd = 'dfRightComposeAffine'
 
-    def _list_outputs(self):
-        outputs = self.output_spec().get()
-        outputs['out_file'] = self.inputs.in_df.replace('.df.nii.gz',
-                                                        '_combo.df.nii.gz')
-        return outputs
-
-
-class diffeoSymTensor3DVolInputSpec(CommandLineInputSpec):
-    in_tensor = traits.Str(desc='moving tensor', exists=True, mandatory=False,
-                           position=0, argstr="-in %s")
-    in_xfm = traits.Str(desc='transform to apply', exists=True,
-                        mandatory=False,
-                        position=1, argstr="-trans %s")
-    in_target = traits.Str(desc='', exists=True, mandatory=False, position=2,
-                           argstr="-target %s")
-    out_file = traits.Str(desc='', exists=True, mandatory=False, position=3,
-                          argstr="-out %s", name_source="in_tensor",
-                          name_template="%s_diffeoxfmd.nii.gz")
-
-
-class diffeoSymTensor3DVolOutputSpec(TraitedSpec):
-    out_file = traits.File(desc='cheese', exists=True)
-
-
-class diffeoSymTensor3DVolTask(CommandLineDtitk):
-    """
-    Applies diffeomorphic transform to a tensor volume
-
-                Example
-                -------
-
-                >>> import nipype.interfaces.dtitk as dtitk
-                >>> node = dtitk.diffeoSymTensor3DVolTask()
-                >>> node.inputs.in_tensor = 'diffusion.nii'
-                >>> node.inputs.in_xfm = 'ants_Warp.nii.gz'
-                >>> node.run() # doctest: +SKIP
-                """
-
-    input_spec = diffeoSymTensor3DVolInputSpec
-    output_spec = diffeoSymTensor3DVolOutputSpec
-    _cmd = 'deformationSymTensor3DVolume'
-
-    def _list_outputs(self):
-        outputs = self.output_spec().get()
-        outputs['out_file'] = self.inputs.out_file
-        return outputs
-
 
 class affSymTensor3DVolInputSpec(CommandLineInputSpec):
-    in_tensor = traits.Str(desc='moving tensor', exists=True, mandatory=False,
-                           position=0, argstr="-in %s")
-    in_xfm = traits.Str(desc='transform to apply', exists=True,
-                        mandatory=False, position=1, argstr="-trans %s")
-    in_target = traits.Str(desc='', exists=True, mandatory=False, position=2,
-                           argstr="-target %s")
-    out_file = traits.Str(desc='', exists=True, mandatory=False, position=3,
-                          argstr="-out %s", name_source="in_tensor",
-                          name_template="%s_affxfmd.nii.gz")
+    in_file = File(desc='moving tensor volume', exists=True,
+                   argstr="-in %s", mandatory=True)
+    out_file = traits.Str(desc='output filename', exists=True,
```

Comment:
This should be a `File` without `exists=True`.

Reply:
I removed `exists=True` for all `out_file` fields in the inputs specs, and changed the `traits.Str` to `traits.File`.

Reply:
Just a note that we use `File` (imported from `nipype.interfaces.base`) not `traits.File`. It has slightly improved error messages compared to `traits.File`.